spec/controllers/hq/partners_controller_spec.rb

```diff
@@ -20,26 +20,52 @@ def foobar_action
 end
 
 RSpec.describe Hq::PartnersController, type: :controller do
-  before :each do
-    sign_in instance_double(AdminUser)
-    @partner = build_stubbed :partner
-    expect(Partner).to receive(:find).and_return(@partner)
-  end
 
-  specify 'editing renders the edit view' do
-    get :edit, id: @partner.id
-    expect(response).to render_template 'edit'
-  end
+  context '#new and #create' do
+    
+    before :each do
+      sign_in instance_double(AdminUser)
+      @partner = FactoryGirl.build(:partner)
+    end
+    
+    specify 'successful create redirects to the show view' do
+      post :create, partner: @partner.attributes
+      @partner = Partner.find_by_name(@partner[:name])
+      expect(response).to redirect_to hq_partner_path(@partner)
+    end
```

Comment:
I think you got it all wrong here. `@partner.save`  IS still touching the database.
And that is why you need the `province_id` and `start_date`, to pass validation. (you could avoid this by stubbing `.valid?` and returning `true` but still `.save` will touch the db).

`@partner` is totally different from `@partner` from controllers! So instead of stubbing `@partner` we need to stubb `Partner.any_instance.save`. (still not enough)
Why is it working for `update` and not for `create`? In update the `@partner = Partner.find(params[:id])` will point to the stubbed object created in the tests (same ids).

Why stubbing `save` when doing the create will not work:
-`params.permit` will not let us send an id to controller, so the newly created @partner will have `:id => nil`
-`@partner.save` : even if it is stubbed and it will return `true`, it will not change the `@partner` instance var (`self` inside `save` method). And inside the NOT stubbed `save` method there is something like `self.id = 1` (this will change `@partner.id`). (objects are sent by reference)

**So in one line:** Stubbing only returns something, it doesn't do any processing like setting and ID to `self` (here @partner).